client/my-sites/comments/comment-list/index.jsx

```diff
@@ -132,18 +146,30 @@ export class CommentList extends Component {
 		} );
 	};
 
-	setCommentStatus = ( commentId, postId, status, options = { showNotice: true } ) => {
+	setCommentStatus = ( commentId, postId, status, options = { isUndo: false, persist: false, showNotice: true } ) => {
+		const { isUndo, persist, showNotice } = options;
+
 		// TODO: Replace with Redux getComment()
 		const comment = this.getComment( commentId );
 
 		if ( comment && status === comment.status ) {
 			return;
 		}
 
+		if ( persist ) {
+			this.updatePersistedComments( {
+				...comment,
+				i_like: 'approved' !== status ? false : comment.i_like,
+				status,
+			}, isUndo );
```

Comment:
let's be careful here. we're actually storing a copy of data that's available in the Redux state tree _and_ we're changing properties on it. we're opening ourselves up to synchronization bugs (for instance, if someone updates the comment in another session and Calypso becomes aware of it, this will still show stale data).

granted this is not a huge risk but I think it's probably also easy to store the list of comment ids that are persisted. in that case (and hopefully if we move to passing lists of ids anyway) the views will always be up to date because they will get their data from state directly and no data will be cloned in independent silos.